File: yote/logs.py
#!/usr/bin/env python3
"""Log aggregation"""
import glob
import os

LOG_DIR = "/home/toxic/sovereign/logs"

def tail_log(name, lines=50):
    path = os.path.join(LOG_DIR, f"{name}.log")
    if not os.path.exists(path):
        return []
    with open(path) as f:
        return f.readlines()[-lines:]

def list_logs():
    return glob.glob(f"{LOG_DIR}/*.log")

File: projects/yote/src/lib/logs.ts
import { Glob } from "bun"
const LD=process.env.YOTE_LOG_DIR||"./logs"
export async function tailLog(n:string,l=50){const p=n.includes("/")?n:`${LD}/${n}.log`;const f=Bun.file(p);if(!(await f.exists()))return[];const t=await f.text();return t.trim().split("\n").slice(-l)}
export function listLogs(){try{const g=new Glob("**/*.log");return Array.from(g.scanSync(LD)).map(f=>`${LD}/${f}`)}catch{return[]}}
